core/src/main/scala/org/apache/spark/SparkEnv.scala

```diff
@@ -402,7 +405,7 @@ object SparkEnv extends Logging {
             None
           }, blockManagerInfo,
           mapOutputTracker.asInstanceOf[MapOutputTrackerMaster],
-          shuffleManager,
+          shuffleBlockGetterFn,
```

Comment:
Why not define `shuffleBlockGetterFn` in `BlockManagerMasterEndpoint`?

Reply:
I see this being an issue in tests where the `SparkEnv` would not be set, so now I'd have to make sure that the env is set and cleared in the tests. That said, if you feel strongly about this, I can look at this more.